### PR TITLE
[VEGA-3140] test(components): Покрытие тестами utils

### PR DESCRIPTION
**Pull request by DmitriiBo**
## Задача
Задача в Jira: https://jira.konakov.tv/browse/VEGA-3140
Ссылка на стенд: https://VEGA-3140.vega-2.csssr.cloud
Jira issue: VEGA-3140

### Description
Тесты на утилитарные функции из директории utils

### Checks
- [x] Jira link attached: [VEGA-3140](https://jira.vegaspace.tk/browse/VEGA-3140)
- [ ] Possible Associated Jira links:
- [ ] Personal Vega Instance link attached: http://xx-yy-vega-builder.code013.org
- [ ] Jira tasks for technical debt were created (if needed)

### Type
- [ ] Feature
- [ ] Alpha Feature ([Alpha] MR/PR Title)
- [ ] Bug Fix
- [x] Test
- [ ] Refactoring

### PR/MR Requirements
- [x] PR/MR title meet requirements/convention
- [x] Target branch is correct
- [ ] PR/MR branch was rebased at correct branch
- [x] Diff was checked
- [ ] Configuration files were checked at test repositories
- [x] [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

### General
- [ ] Exploratory Tested at Personal Vega Instance
- [ ] Existing E2E Tests done, Screenshot attached
- [ ] Existing Integration Tests done, Screenshot attached, Jenkins link:
- [x] Existing Unit Tests done, Screenshot attached
- [ ] New E2E Tests developed
- [ ] New Integration Tests developed
- [x] New Unit Tests developed
- [ ] API documentation was fully and correctly updated

### E2E Suite Jenkins link
_{link}_

### E2E Suite Screenshot
_{Screenshot}_

### Integration Tests Screenshot
_{Screenshot}_

### Unit Tests Screenshot
_{Screenshot}_
image

### CODE REVIEW

File: __test__/utils/clear-array-from-duplicates.test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const inputArray = [1, 1, 2, 3, 3];
+
+    const resultArray = clearArrayFromDuplicates(inputArray);
+
+    expect(resultArray).toEqual([1, 2, 3]);
+  });
+
+  test('Удаляет дубликаты объектов', () => {
+    const object = { test: 'test' };
+    const duplicatedObject = object;
```
Comment (lipgen) [inline __test__/utils/clear-array-from-duplicates.test.ts:22]: а здесь он должен проверить ссылки на один и тот же объект, или два объекта с идентичным содержимым? Если второе, то мне кажется стоит присваивать ```{ ...object }``` или еще раз ```{ test: 'test' }```

Comment (DmitriiBo) [reply __test__/utils/clear-array-from-duplicates.test.ts:22]: Тут проверка на идентичность ссылок

File: __test__/utils/clear-array-from-duplicates.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    const resultArray = clearArrayFromDuplicates(inputArray);
+
+    expect(resultArray.length).toBe(1);
+    expect(resultArray.find((obj) => obj.test === 'test')).toBe(duplicatedObject);
+  });
+
+  test('Удаляет дубликаты массивов', () => {
+    const array = ['test'];
+    const duplicatedArray = array;
```
Comment (lipgen) [inline __test__/utils/clear-array-from-duplicates.test.ts:34]: тоже самое, что и с объектами

Comment (DmitriiBo) [reply __test__/utils/clear-array-from-duplicates.test.ts:34]: Выше отписал

**Review by lipgen — APPROVED** [review]

File: __test__/utils/clear-array-from-duplicates.test.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { clearArrayFromDuplicates } from '../../src/utils/clear-array-from-duplicates';
+
+describe('Функция удаления дублей из массивов', () => {
+  test('Удаляет дубликаты строк', () => {
+    const inputArray = ['1', '1', '2', '3'];
```
Comment (ababakov) [inline __test__/utils/clear-array-from-duplicates.test.ts:5]: параметризация тестов думаю должна быть https://medium.com/@suman.maity112/parameterized-tests-with-jest-2932f42b1509

**Review by ababakov — APPROVED** [review]

**Pull request merged**